### PR TITLE
Break transformText into new functions that can be exported

### PR DESCRIPTION
**Pull request by fsih**
Depends on https://github.com/LLK/scratch-render-fonts/pull/3
- Breaks apart transform text, and exposes some useful functions that will be used in the dependent PRs
- Adds convertFont, which converts 2.0 to 3.0 fonts.

### CODE REVIEW

**Comment by fsih** [comment]
There's an svg like this in project 221549537:
`<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="898.364" height="399.7798" shape-rendering="crispEdges"><g transform="translate(2,47.70141)"><g data-paper-data="{&quot;isPaintingLayer&quot;:true}" fill="none" fill-rule="nonzero" stroke="none" stroke-width="2" stroke-linecap="butt" stroke-linejoin="miter" stroke-miterlimit="10" stroke-dasharray="" stroke-dashoffset="0" font-family="Typey McTypeface" font-weight="400" font-size="22" text-anchor="start" style="mix-blend-mode: normal"><g><text transform="translate(-2,125.2) scale(8.73239,8.73239)" id="ID0.5187592278234661" fill="#24311c"><tspan x="0" dy="0">Kilorush</tspan></text><text transform="translate(28,326.5) scale(3.87551,3.87551)" id="ID0.23731968319043517" fill="#ededed"><tspan x="0" dy="0">Into the Clouds</tspan></text><text transform="translate(676,297) scale(7.67552,7.67552)" id="ID0.8837321638129652" fill="#24311c"><tspan x="0" dy="0">3D</tspan></text></g></g></g></svg>`

The text elements don't have font face defined directly on them. Instead the font is defined on a tag which contains several text elements.
The current transform text method doesn't handle this case. The fonts will be lost when converting to image.

**Comment by fsih** [comment]
I rewrote the font family finder code to fix the thing I commented about

File: src/font-inliner.js
Comment (kchadha) [inline src/font-inliner.js]: Does it make sense to use a Set instead?

File: src/font-inliner.js
Comment (kchadha) [inline src/font-inliner.js]: If you use the suggestion from the previous comment, you can replace the above two lines with `for (let font of fontsNeeded) {`

**Review by kchadha — COMMENTED** [review]
Minor suggestion in the comments, but other than that this looks okay to me! You might want to wait for a final review from @cwillisf, though.

File: src/font-inliner.js
Comment (fsih) [reply src/font-inliner.js]: Now that you mention it, I'm not sure why transformText uses an object for fontsNeeded, and not say an array. Maybe it's kind of like having a typed array of strings?

I think I'll leave it as it was, unless you feel it's particularly confusing?

**Comment by fsih** [comment]
Huzzah finally fixed that dep

**Comment by fsih** [comment]
Thanks @kchadha!
@cwillisf do you mind taking a look?

File: src/svg-renderer.js
Comment (cwillisf) [inline src/svg-renderer.js]: Super nit-pick, but I would prefer Boolean arguments to be phrased in terms of what it means when the value is true. In this case, if the value is true it means that this code should inject the fonts -- not that they've already been injected. I would call it something like `shouldInjectFonts` or `fontInjectionRequested` or something like that.

File: src/font-inliner.js
Comment (cwillisf) [reply src/font-inliner.js]: I think a `Set` would make sense here -- in fact, I think the original code is using an object as a string-to-Boolean map / dictionary, which they're using to emulate a `Set`. I don't think it's critical to change it right this moment, but it would express the intent better and simplify the loop below as @kchadha observed.

File: src/font-converter.js
Comment (cwillisf) [inline src/font-converter.js]: I feel really weird about replacing the `font-family` in this way -- especially for something as generic as "Helvetica" -- but I guess it's OK as long as this only runs on Scratch 2.0 SVGs. I do think we should resolve the names as soon as we can ;)

**Review by cwillisf — APPROVED** [review]

File: src/font-converter.js
Comment (fsih) [reply src/font-converter.js]: Names resolved!

There's going to be an intermediate period where the renderer still uses svgRenderer, so if you import a costume in any way (not just from 2.0), it will convert helvetica to noto, and the paint editor won't (Though I'm not really sure what the paint editor UI should look like if it encounters fonts it doesn't have, hmm)

File: src/font-inliner.js
Comment (fsih) [reply src/font-inliner.js]: Done

File: src/font-inliner.js
Comment (fsih) [reply src/font-inliner.js]: Done

File: src/svg-renderer.js
Comment (fsih) [reply src/svg-renderer.js]: No problem, I think that makes sense

**Pull request merged**
